lib/systemd/journal.rb

```diff
@@ -46,6 +46,7 @@ class Journal
     #     path: '/var/log/journal/5f5777e46c5f4131bd9b71cbed6b9abf'
     #   )
     def initialize(opts = {})
+      @reopen_options = opts.dup # retain the options for auto reopen
```

Comment:
can we add a `auto_reopen:` (or something) optional parameter here? I think this change would be best as an opt-in feature, at least to start.

```ruby
Journal.new(auto_reopen: false) # do not ever re-open the journal.
                                # this should be the default for now

Journal.new(auto_reopen: true)   # re-open the journal after the default LIMIT_TO_AUTO_REOPEN
Journal.new(auto_reopen: 50_000) # re-open the journal after 50k iterations
```

Reply:
Thank you for your quick response.
I will try to fix your comment at tomorrow.

Reply:
Added the feature as opt-in
https://github.com/ledbettj/systemd-journal/pull/103/commits/ee018d4c33ec5665940754b37ad239d1a371a1c8

Thanks